packages/react-admin/src/sideEffect/saga/i18nSaga.js

```diff
@@ -0,0 +1,68 @@
+import { parse } from 'query-string';
+import {
+    CHANGE_LOCALE,
+    USER_LOGOUT,
+    changeMessages,
+    changeLocale,
+} from '../../actions';
+import { all, call, put, select, takeLatest } from 'redux-saga/effects';
+import { LOCATION_CHANGE } from 'react-router-redux';
+
+const resolveMessages = messages =>
+    Promise.resolve(typeof messages === 'function' ? messages() : messages);
+const localeSelector = state => state.i18n.locale;
+
+export default ({
+    locale: initialLocale,
+    messages,
```

Comment:
Storing the messages in the saga feels weird to me. I can't put the finger on why - probably because usually, sagas don't persist anything. storing stuff is the store's job.